shared-lib/spiutils/src/protocol/flash.rs

```diff
@@ -208,6 +209,26 @@ impl<'a> OpCode {
 
 const DUMMY_BYTE_VALUE: u8 = 0xff;
 
+/// Address in SPI flash protocol header
+pub trait Address: BeInt + Into::<u32> {
+    /// Convert from a u32.
+    /// Note: This will become redundant once types from `ux` crate implement
+    /// `core::convert::TryFrom`.
+    fn try_from(val: u32) -> Result<Self, core::num::TryFromIntError>;
+}
+
+impl Address for ux::u24 {
+    fn try_from(val: u32) -> Result<Self, core::num::TryFromIntError> {
+        Ok(ux::u24::new(val))
```

Comment:
I would expect this to return an error if `val > u24::max_value()`. Does this need clarification or fixing?

Reply:
Oh, that was a good call! On inspection, I found that ux::[type]::new just asserts that the value fits and panics if it doesn't. I replaced the TryFromIntError with a custom error since TryFromIntError cannot be instantiated from outside the crate that it's defined in.